src/devices/GrovePi/GrovePi.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.GrovePiDevice.Models;
+using System;
+using System.Buffers.Binary;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.IO;
+using System.Threading;
+
+namespace Iot.Device.GrovePiDevice
+{
+    /// <summary>
+    /// Create a GrovePi class
+    /// </summary>
+    public class GrovePi : IDisposable
+    {
+        private I2cDevice _i2cDevice;
+        private readonly bool _autoDispose;
+        private const byte MaxRetries = 4;
+
+        /// <summary>
+        /// The default GrovePi I2C address is 0x04
+        /// Other addresses can be use, see GrovePi documentation
```

Comment:
NIT: I wonder if it would be worth it to add here the alternative I2c addresses as well. It might be nice so that users don't have to look into datasheet in case they want to use an alternate address. Anyways, you don't have to fix it, just thinking out loud.

Reply:
the way this device is built it's rather unlikely anyone will want to change this (basically there is only one reasonable way to connect it)

Reply:
There is a way to change the address programmatically thru some scripts. To implement that, it would need to add SPI support and basically do what the script is doing. I Don't see much interest. Usually what people do is changing it before they start using it. If they want to have multiple GrovePi, they need anyway to change one by one with the script.